optimism/material/MaterialModel.py

```diff
@@ -4,5 +4,5 @@
 
 MaterialModel = namedtuple('MaterialModel',
                            ['compute_energy_density', 'compute_initial_state', 'compute_state_new',
-                            'density'],
+                            'density','compute_material_qoi'],
                            defaults=(0.0,))
```

Comment:
I think this is a bug. The default values for `namedtuple` are applied from the right, so I think this will now default the value for `compute_material_qoi` to 0.0 instead of `density`. I think you should put  `compute_material_qoi` before `density`?